Python_Code_Tiktok_Chat_To_OSC

```diff
@@ -3,73 +3,66 @@ from pythonosc import udp_client
 from TikTokLive import TikTokLiveClient
 from TikTokLive.events import GiftEvent, CommentEvent
 
-# Variables to track the state of roses, galaxies, and crowns
-rose = 0
-galaxy = 0
-boop_toggle = 0
+class TikTokEventHandler:
+    def __init__(self, ip: str, port: int, username: str):
+        self.rose = 0
+        self.galaxy = 0
+        self.boop_toggle = 0
+        self.client = udp_client.SimpleUDPClient(ip, port)
+        self.tiktok_client = TikTokLiveClient(unique_id=username)
 
+        # Register event handlers
+        self.tiktok_client.on(GiftEvent, self.handle_gift)
+        self.tiktok_client.on(CommentEvent, self.handle_comment)
 
-# Function to send a request to the IP address with the current state of roses, galaxies, crowns, or boop
-def send_request(parameter, value):
-    client = udp_client.SimpleUDPClient("127.0.0.1", 9000)
-    client.send_message("/avatar/parameters/" + parameter, value)
+    def send_request(self, parameter: str, value: int):
+        # Send a request to the IP address with the current state of the parameter.
+        self.client.send_message(f"/avatar/parameters/{parameter}", value)
 
+    async def handle_gift(self, event: GiftEvent):
+        # Handle gift events and send appropriate requests.
+        if event.gift.name == "Rose":
+            self.rose = 1
+            self.send_request("rose", 1)
+            await asyncio.sleep(2)
+            self.rose = 0
+            self.send_request("rose", 0)
+        elif event.gift.name == "Galaxy":
+            self.galaxy = 1
+            self.send_request("galaxy", 1)
+            await asyncio.sleep(2)
+            self.galaxy = 0
+            self.send_request("galaxy", 0)
```

Comment:
If the event names are always known and in the requests, it could probably be simplified to
```
allowed_events = ["Rose", "Galaxy"]
event_name = event.gift.name
if event_name in allowed_events:
  self.send_request(event_name.lower(), 1)
  await asyncio.sleep(2)
  self.send_request(event_name.lower(0))
```

Assuming my previous comment is accurate and the `self.rose = 1` & `self.rose = 0` does nothing